asdf/_display.py

```diff
@@ -92,9 +98,13 @@ def from_root_node(cls, root_identifier, root_node):
                     if parent is not None:
                         parent.children.append(info)
                     seen.add(id(node))
-
-                    for child_identifier, child_node in get_children(node):
-                        next_nodes.append((info, child_identifier, child_node))
+                    if cls.supports_info(node):
+                        info.tag = node._tag
+                        for child_identifier, child_node in node.__asdf_traverse__():
```

Comment:
Something I missed before, if we call `get_children` on the result of `__asdf_traverse__`, that will allow `__asdf_traverse__` to return a dict and have it processed correctly.

Reply:
I'm not sure what you mean here, particularly with _if we call_ `get_children` on the result. Doesn't `__asdf_traverse__` return the same kind of result that `get_children` does? I may be slow this morning. It is Wednesday.

Reply:
The way this is set up currently, the `__asdf_traverse__` method must return dict.items() if it wants to provide dict items, and enumerate(list) if it wants to provide list items.  If we instead call `get_children(__asdf_traverse__())` here, then `__asdf_traverse__` can return simple dict or list and have everything work correctly (get_children will take care of calling .items() on the dict and wrapping the list in enumerate).

Reply:
OK, now I understand. I didn't realize you wanted to change what `__asdf_traverse__` should return, I'll make the corresponding changes. It would be a simpler requirement on the method.